arches/app/etl_modules/base_data_editor.py

```diff
@@ -129,73 +132,68 @@ def edit_staged_data(self, cursor, graph_id, node_id, operation, language_code,
             result["message"] = _("Unable to edit staged data: {}").format(str(e))
         return result
 
-    def get_preview_data(self, graph_id, node_id, resourceids, language_code, old_text, case_insensitive):
-        node_id_query = " AND nodeid = %(node_id)s" if node_id else ""
-        graph_id_query = " AND graphid = %(graph_id)s" if graph_id else ""
-        resourceids_query = " AND resourceinstanceid IN %(resourceids)s" if resourceids else ""
-        like_operator = "ilike" if case_insensitive == "true" else "like"
-        old_text_like = "%" + old_text + "%" if old_text else ""
-        text_query = (
-            " AND t.tiledata -> %(node_id)s -> %(language_code)s ->> 'value' " + like_operator + " %(old_text)s" if old_text else ""
-        )
+    def get_preview_data(self, graph_id, node_id, search_url, language_code, old_text, case_insensitive):
+        request = HttpRequest()
+        request.user = self.request.user
+        request.method = "GET"
+        request.GET["paging-filter"] = 1
+        request.GET["tiles"] = True
+
         if language_code is None:
             language_code = "en"
 
-        request_parmas_dict = {
-            "node_id": node_id,
-            "language_code": language_code,
-            "graph_id": graph_id,
-            "resourceids": resourceids,
-            "old_text": old_text_like,
-        }
+        if search_url:
+            params = parse_qs(urlsplit(search_url).query)
+            for k, v in params.items():
+                request.GET.__setitem__(k, v[0])
+
+        search_url_query = search_results(request, returnDsl=True).dsl["query"]
+        case_insensitive = True if case_insensitive == "true" else False
+
+        if old_text:
+            node_value_query = Wildcard(
+                field=f"tiles.data.{node_id}.{language_code}.value.keyword",
+                query=f"*{old_text}*",
+                case_insensitive=case_insensitive,
+            )
+            string_search_nested = Nested(path="tiles", query=node_value_query)
```

Comment:
This extra Nested filter is redundant and seems to fix the count of tiles issue for search and replace.